config/initializers/spree.rb

```diff
@@ -38,7 +57,39 @@
   config.s3_access_key = ENV['S3_ACCESS_KEY'] if ENV['S3_ACCESS_KEY']
   config.s3_secret = ENV['S3_SECRET'] if ENV['S3_SECRET']
   config.use_s3 = true if ENV['S3_BUCKET']
+  config.s3_headers = ENV['S3_HEADERS'] if ENV['S3_HEADERS']
   config.s3_protocol = ENV.fetch('S3_PROTOCOL', 'https')
+
+  # Attachments settings
+  config.attachment_default_url = ENV['ATTACHMENT_DEFAULT_URL'] if ENV['ATTACHMENT_DEFAULT_URL']
+  config.attachment_path = ENV['ATTACHMENT_PATH'] if ENV['ATTACHMENT_PATH']
+  config.attachment_url = ENV['ATTACHMENT_URL'] if ENV['ATTACHMENT_URL']
+  config.attachment_styles = ENV['ATTACHMENT_STYLES'] if ENV['ATTACHMENT_STYLES']
+  config.attachment_default_style = ENV['ATTACHMENT_DEFAULT_STYLE'] if ENV['ATTACHMENT_DEFAULT_STYLE']
+
+  # Update paperclip settings
+  if config.use_s3
+    s3_creds = { access_key_id: config.s3_access_key,
+                 secret_access_key: config.s3_secret,
+                 bucket: config.s3_bucket }
+    Spree::Image.attachment_definitions[:attachment][:storage] = :s3
+    Spree::Image.attachment_definitions[:attachment][:s3_credentials] = s3_creds
+    Spree::Image.attachment_definitions[:attachment][:s3_headers] =
+      ActiveSupport::JSON.decode(config.s3_headers)
+    Spree::Image.attachment_definitions[:attachment][:bucket] = config.s3_bucket
+  else
+    Spree::Image.attachment_definitions[:attachment].delete :storage
+  end
+
+  Spree::Image.attachment_definitions[:attachment][:styles] =
+    ActiveSupport::JSON.decode(config.attachment_styles).symbolize_keys!
+  Spree::Image.attachment_definitions[:attachment][:path] = config.attachment_path
+  Spree::Image.attachment_definitions[:attachment][:default_url] =
+    config.attachment_default_url
+  Spree::Image.attachment_definitions[:attachment][:default_style] =
+    config.attachment_default_style
```

Comment:
What about bringing in the image model from spree and define the attachments as usual for Paperclip? The first place I'd go find the settings is in the model because that's the convention. Also, I find tedious having to deal with hashes here instead of the `has_attached_file` DSL, and is not easy to follow either IMO.

Reply:
I'd have to investigate how do it, change the code and test it.
This way I am keeping the code as is (this is exactly what was on the controller) and just moving it around. I see changing code a bit out of scope for this PR. It would be more effort and risk. I'd prefer to go forward as is so that we save dev time. What do you think?